readthedocs/oauth/models.py

```diff
@@ -171,7 +171,7 @@ class RemoteRepository(TimeStampedModel):
         blank=True,
     )
     # VCS provider repository id
-    remote_id = models.CharField(db_index=True, max_length=128)
+    remote_id = models.CharField(max_length=128)
```

Comment:
Hrm, this called my attention. I think we perform a few queries based on `remote_id` in .com because of SSO. Have you checked this index there?